gulp/tasks/lint.js

```diff
@@ -4,16 +4,21 @@ var gulp = require( 'gulp' );
 var plugins = require( 'gulp-load-plugins' )();
 var configLint = require( '../config' ).lint;
 var handleErrors = require( '../utils/handle-errors' );
+var minimist = require( 'minimist' );
 
 /**
  * Generic lint a script source.
  * @param {string} src The path to the source JavaScript.
  * @returns {Object} An output stream from gulp.
  */
 function _genericLint( src ) {
-  return gulp.src( src )
-    .pipe( plugins.eslint() )
+  // Grab the --fix flag from the command-line, if available.
+  var commandLineParams = minimist( process.argv.slice( 2 ) );
+  var willFix = commandLineParams.fix || false;
+  return gulp.src( src, { base: './' } )
```

Comment:
The `base` property is used to set the base path of the source file glob. By setting it to the current directory, and then also providing the same setting in `gulp.dest`, each source file path is used in the destination, effectively overwriting the files. I wasn't sure how else to pass in the destination file paths, which differ widely.